### ISSUE TITLE
Error with composite models in ipython

### ISSUE THREAD

**Issue by tritemio** [lmfit/lmfit-py#349]
Executing the folllowing script in an ipython kernel (either from the terminal or the notebook) gives and error:

```
import lmfit
print('lmfit: %s' % lmfit.__version__)

peak1 = lmfit.models.GaussianModel(prefix='p1_')
peak2 = lmfit.models.GaussianModel(prefix='p2_')
model = peak1 + peak2

model.make_params()
```

Error:
```
lmfit: 0.9.4rc1-7-gae7884b
ZeroDivisionError
 <_ast.Module object at 0x1198cb9e8>
 ^^^
float division by zero
---------------------------------------------------------------------------
ZeroDivisionError Traceback (most recent call last)
/Users/anto/miniconda3/lib/python3.5/site-packages/IPython/core/formatters.py in __call__(self, obj)
 668 type_pprinters=self.type_printers,
 669 deferred_pprinters=self.deferred_printers)
--> 670 printer.pretty(obj)
 671 printer.flush()
 672 return stream.getvalue()
/Users/anto/miniconda3/lib/python3.5/site-packages/IPython/lib/pretty.py in pretty(self, obj)
 366 if cls in self.type_pprinters:
 367 # printer registered in self.type_pprinters
--> 368 return self.type_pprinters[cls](obj, self, cycle)
 369 else:
 370 # deferred printer
/Users/anto/miniconda3/lib/python3.5/site-packages/IPython/lib/pretty.py in _ordereddict_pprint(obj, p, cycle)
 817 p.text('...')
 818 elif len(obj):
--> 819 p.pretty(list(obj.items()))
 820 
 821 def _deque_pprint(obj, p, cycle):
/Users/anto/miniconda3/lib/python3.5/site-packages/IPython/lib/pretty.py in pretty(self, obj)
 366 if cls in self.type_pprinters:
 367 # printer registered in self.type_pprinters
--> 368 return self.type_pprinters[cls](obj, self, cycle)
 369 else:
 370 # deferred printer
/Users/anto/miniconda3/lib/python3.5/site-packages/IPython/lib/pretty.py in inner(obj, p, cycle)
 550 p.text(',')
 551 p.breakable()
--> 552 p.pretty(x)
 553 if len(obj) == 1 and type(obj) is tuple:
 554 # Special case for 1-item tuples.
/Users/anto/miniconda3/lib/python3.5/site-packages/IPython/lib/pretty.py in pretty(self, obj)
 366 if cls in self.type_pprinters:
 367 # printer registered in self.type_pprinters
--> 368 return self.type_pprinters[cls](obj, self, cycle)
 369 else:
 370 # deferred printer
/Users/anto/miniconda3/lib/python3.5/site-packages/IPython/lib/pretty.py in inner(obj, p, cycle)
 550 p.text(',')
 551 p.breakable()
--> 552 p.pretty(x)
 553 if len(obj) == 1 and type(obj) is tuple:
 554 # Special case for 1-item tuples.
/Users/anto/miniconda3/lib/python3.5/site-packages/IPython/lib/pretty.py in pretty(self, obj)
 381 if callable(meth):
 382 return meth(obj, self, cycle)
--> 383 return _default_pprint(obj, self, cycle)
 384 finally:
 385 self.end_group()
/Users/anto/miniconda3/lib/python3.5/site-packages/IPython/lib/pretty.py in _default_pprint(obj, p, cycle)
 501 if _safe_getattr(klass, '__repr__', None) not in _baseclass_reprs:
 502 # A user-provided repr. Find newlines and replace them with p.break_()
--> 503 _repr_pprint(obj, p, cycle)
 504 return
 505 p.begin_group(1, '<')
/Users/anto/miniconda3/lib/python3.5/site-packages/IPython/lib/pretty.py in _repr_pprint(obj, p, cycle)
 692 """A pprint that just redirects to the normal repr function."""
 693 # Find newlines and replace them with p.break_()
--> 694 output = repr(obj)
 695 for idx,output_line in enumerate(output.splitlines()):
 696 if idx:
/Users/anto/src/lmfit-py/lmfit/parameter.py in __repr__(self)
 548 if self.name is not None:
 549 s.append("'%s'" % self.name)
--> 550 sval = repr(self._getval())
 551 if not self.vary and self._expr is None:
 552 sval = "value=%s (fixed)" % sval
/Users/anto/src/lmfit-py/lmfit/parameter.py in _getval(self)
 638 if not self._delay_asteval:
 639 self.value = self._expr_eval(self._expr_ast)
--> 640 check_ast_errors(self._expr_eval)
 641 
 642 v = self._val
/Users/anto/src/lmfit-py/lmfit/parameter.py in check_ast_errors(expr_eval)
 20 """check for errors derived from asteval"""
 21 if len(expr_eval.error) > 0:
---> 22 expr_eval.raise_exception(None)
 23 
 24 
/Users/anto/src/lmfit-py/lmfit/asteval.py in raise_exception(self, node, exc, msg, expr, lineno)
 165 except:
 166 exc = RuntimeError
--> 167 raise exc(self.error_msg)
 168 
 169 
ZeroDivisionError: in expr='<_ast.Module object at 0x1198cb9e8>'
```

No error with plain python, so the nose tests don't catch this.

I'm using IPython 5 and OS X.

**Comment by tritemio** [comment]
I get the same error with ipython 4.2.1.

**Comment by tritemio** [comment]
Same error with conda's ipython 4.2.0.

attn: @takluyver

**Comment by tritemio** [comment]
Same error with lmfit 0.9.3 (all previous tests were done with current master).

**Comment by newville** [comment]
@tritemio @takluyver  I am almost certain this is a lmfit problem, and has nothing to do with IPython.

Without delving too deeply into this,

```python 
model = lmfit.models.SomeModel()
pars = model.make_params()
```
(that is, without providing any initial values for the Parameters) will lead to parameters without "good" initial values, say 0 or +/Inf.      That could easily lead to a ZeroDivisionError or other problems if those parameters are actually used.

For GaussianModel and other standard peak-like models, the derived "height" parameter is
``` 
factor * amplitude / sigma
```
Leaving the value for the `sigma` parameter unspecified could easily cause the problem.  I am sure

```python
pars = model.make_params(p1_sigma=0.001, p2_sigma=0.002)
```
will avoid the problem.  

Perhaps the ZeroDivision should be suppressed in the definition of `height`?  

We used to have unspecified parameter values be `None`, but that caused many troubles too. 

Of course, to be useful, Parameter values do have to be specified.

**Comment by tritemio** [comment]
I see, thanks for the clarification.

Why don't we default to some sensible parameters for out built-in models, let's say 0 for location parameters and 1 for all the others (amplitude, sigma, etc...)?

**Comment by newville** [comment]
@tritemio 

>Why don't we default to some sensible parameters for out built-in models, let's say 0 for 
> location  parameters and 1 for all the others (amplitude, sigma, etc...)?

For a simple model, this actually does happen (taking default values from the signature of the underlying function).  This must not be working with composite models.   Needs investigation!

**Comment by newville** [comment]
should be addressed with #351
